src/include/duckdb/parser/parsed_data/create_type_info.hpp

```diff
@@ -15,16 +15,22 @@
 
 namespace duckdb {
 
+//! The type to bind type modifiers to a type
+typedef void (*bind_type_modifiers_function_t)(ClientContext &context, LogicalType &type,
```

Comment:
Let's make this take a `const LogicalType &type` and return a `LogicalType` instead